agent/util-scripts/pbench-register-tool-set

```diff
@@ -129,6 +129,63 @@ if [[ -z ${the_tool_set} ]]; then
 	exit 1
 fi
 
+# this code is a cut-and-paste from pbench-register-tool
+# If we are given a remotes file, check it for sanity now,
+# before trying to call pbench-register-tool and failing (multiple times).
+if [[ ${remotes_arg::1} == "@" ]]; then
+	# We are dealing with a file containing the list of remotes.
+
+	if [[ -n ${labels_arg} ]]; then
+		error_log "--labels=${labels_arg} not allowed with remotes file (${remotes_arg})"
+		usage >&2
+		exit 1
+	fi
+	remotes_file=${remotes_arg#*@}
+	if [[ ! -e ${remotes_file} ]]; then
+		error_log "--remotes=@${remotes_file} specifies a file that does not exist"
+		usage >&2
+		exit 1
+	fi
+	declare -a remotes_A
+	declare -A labels_A
+	idx=0
+	lineno=0
+	while read line; do
+		if [[ ${line::1} == "#" ]]; then
+			# Ignore comments
+			(( lineno++ ))
+			continue
+		fi
+		IFS=',' read -ra parts <<< "${line}"
+		if [[ ${#parts[@]} == 0 ]]; then
+			# Ignore empty lines
+			(( lineno++ ))
+			continue
```

Comment:
I really think this makes it harder to read and the added generality is not worth it. When it's rewritten in Python, we can worry about generalizing comments: at least we'll have a more expressive language to do it in.

Reply:
I'd rather see this change made in a separate PR, where we handle "empty" remotes files as well.

Reply:
This is actually already taken care of: when the line is read into the array, white space in an otherwise empty line does the same thing as a completely empty line.  Create a file `foo` with one line consisting solely of white space. Then run this:

```
 $ unset parts; while read line ;do IFS=',' read -ra parts <<< ${line}; echo ${#parts[@]};done < foo
0
$
```
It falls into the empty-line category.

SInce I got this for free, I did what @webbnh suggested for the comment case, so these can be put to bed.